src/pymodaq_plugins_signal_recovery/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Lockin_DSP7265.py

```diff
@@ -0,0 +1,187 @@
+                labels = child.value()['selected']
+                data.append(
+                    DataFromPlugins(
+                        name=child.name(),
+                        data=[np.array([0]) for _ in labels],
+                        labels=labels,
+                        dim='Data0D'
+                    )
+                )
+            self.data_grabed_signal_temp.emit(data)
+
+    def ini_detector(self, controller: object = None) -> edict:
+        """Viewer communication initialization
+
+        Parameters
+        ----------
+        controller: (object, optional)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one
+            actuator by controller (Master case)
+
+        Returns
+        -------
+        info: str
+            Id of the lockin
+        initialized: bool
+            False if initialization failed otherwise True
+        controller: object
+            Controller of the daq_viewer
+        """
+        try:
+            self.status.update(
+                edict(info="", controller=None, initialized=False)
+            )
+            if self.settings.child('controller_status').value() == "Slave":
+                if controller is None:
+                    raise Exception('No controller has been defined externally'
+                                    'while this axe is a slave one')
+                else:
+                    print(controller)
+                    self.controller = controller
+            else:
+                adapter = ADAPTERS[self.settings.child('adapter').value()](
+                    self.settings.child('address').value()
+                )
+                self.controller = DSP7265ThreadSafe(adapter)
+
+            self.status.info = self.controller.id
+            self.status.controller = controller
+            self.status.initialized = True
+            return self.status
+
+        except Exception as e:
+            self.emit_status(
+                ThreadCommand('Update_Status', [getLineInfo() + str(e), 'log'])
+            )
+            self.status.info = getLineInfo() + str(e)
+            self.status.initialized = False
+            return self.status
+
+    def stop(self) -> None:
+        return ""
+
+    def close(self) -> None:
+        pass
+
+    def grab_data(self, Naverage: int = 1, **kwargs) -> None:
+        """Grab data
+
+        Args:
+            Naverage (int, optional): Number of averaging if available.
+            Defaults to 1.
+        """
+        data = []
+        for child in self.settings.child('channels').children():
+            labels = child.value()['selected'][:]
+            subdata = [np.array([getattr(self.controller, label)])
+                       for label in labels]
+            data.append(DataFromPlugins(
+                name=child.name(),
+                data=subdata,
+                labels=labels,
+                dim='Data0D'
+            ))
+        self.data_grabed_signal.emit(data)
```

Comment:
look at the template and use dte_signal to emit your data. These are DataToExport object containing DataFromPlugins objects. Look at the plugin template for exemples!